messaging/app/src/main/java/com/google/firebase/example/messaging/kotlin/MainActivity.kt

```diff
@@ -90,4 +96,40 @@ class MainActivity : AppCompatActivity() {
         // [END log_reg_token]
     }
 
+    @RequiresApi(33)
+    // [START ask_post_notifications]
+    private fun askNotificationPermission() {
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.POST_NOTIFICATIONS) ==
+            PackageManager.PERMISSION_GRANTED
+        ) {
+            // FCM SDK (and your app) can post notifications.
+        } else if (shouldShowRequestPermissionRationale(Manifest.permission.POST_NOTIFICATIONS)) {
+            // TODO: display an educational UI explaining to the user the features that will be enabled
+            //       by them granting the POST_NOTIFICATION permission. This UI should provide the user
```

Comment:
Same comment as the Android version here regarding commas.

Reply:
Done!